test/unit/common/test_index2d.cpp

```diff
@@ -59,18 +59,6 @@ TYPED_TEST(Index2DTest, ConstructorFromArray) {
   EXPECT_TRUE(index.isValid());
 }
 
-TYPED_TEST(Index2DTest, Negative) {
-  auto index_tests =
-      std::vector<std::pair<TypeParam, TypeParam>>{std::make_pair(-10, -10), std::make_pair(-1, -1),
-                                                   std::make_pair(-15, 3), std::make_pair(8, -7)};
-
-  for (auto& test : index_tests) {
-    EXPECT_THROW(Index2D<TypeParam>(test.first, test.second), std::invalid_argument);
-    EXPECT_THROW(Index2D<TypeParam>(std::array<TypeParam, 2>{test.first, test.second}),
-                 std::invalid_argument);
-  }
-}
-
```

Comment:
You should keep this tests:
You should check that you can construct objects with negative indices, that the values are correctly set, and then that is_valid() on those objects returns false.